src/metabase/search/appdb/index.clj

```diff
@@ -98,7 +98,7 @@
        (t2/query {:select [:table_name]
                   :from   :information_schema.tables
                   :where  [:and
-                           [:= :table_schema :%current_schema]
+                           [:= :table_schema (if (= (mdb/db-type) :mysql) :%schema :%current_schema)]
```

Comment:
Not worth abstracting into specialization code? Could simply be a constant for the keyword.